examples/blake2f.rs

```diff
@@ -0,0 +1,1377 @@
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vd_bit_vec = v_mid_vd_bit_vec.clone();
+                let v_mid_vc_bit_vec: Vec<Vec<Queriable<F>>> = (0..V_LEN)
+                    .map(|i| {
+                        (0..BITS_NUMBER + 1)
+                            .map(|j| {
+                                ctx.internal(format!("v_mid_vc_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vc_bit_vec = v_mid_vc_bit_vec.clone();
+                let v_mid_vb_bit_vec: Vec<Vec<Queriable<F>>> = (0..V_LEN)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_mid_vb_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_mid_vb_bit_vec = v_mid_vb_bit_vec.clone();
+
+                let v_xor_d_bit_vec: Vec<Vec<Queriable<F>>> = (0..G_ROUND)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_xor_d_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_xor_d_bit_vec = v_xor_d_bit_vec.clone();
+
+                let v_xor_b_bit_vec: Vec<Vec<Queriable<F>>> = (0..G_ROUND)
+                    .map(|i| {
+                        (0..BITS_NUMBER)
+                            .map(|j| {
+                                ctx.internal(format!("v_xor_b_bit_vec[{}][{}]", i, j).as_str())
+                            })
+                            .collect()
+                    })
+                    .collect();
+                let setup_v_xor_b_bit_vec = v_xor_b_bit_vec.clone();
+
+                let b_bit_vec: Vec<Queriable<F>> = (0..G_ROUND / 2)
+                    .map(|i| ctx.internal(format!("b_bit_vec[{}]", i).as_str()))
+                    .collect();
+                let setup_b_bit_vec = b_bit_vec.clone();
+                let b_3bits_vec: Vec<Queriable<F>> = (0..G_ROUND / 2)
+                    .map(|i| ctx.internal(format!("b_3bits_vec[{}]", i).as_str()))
+                    .collect();
+                let setup_b_3bits_vec = b_3bits_vec.clone();
+
+                ctx.setup(move |ctx| {
+                    let s = SIGMA_VALUES[r % 10];
+
+                    // Rotation constants | (R1, R2, R3, R4)  | = | (32, 24, 16, 63)
+                    for i in 0..G_ROUND {
+                        let mut input_vec = v_vec.clone();
```

Comment:
@10to4 I think the content of this loop is too long, perhaps you should extract it to a function called g_setup or something like that, that receives a borrow of the context and the signals. Perhaps rename `g_internal` to `g_wg` or `g_witness`